File: sdk/typescript-sdk/examples/basic-compression.ts
/**
 * Basic compression example
 */

import { Concise } from 'concise-sdk';

const client = new Concise({ apiKey: 'your-api-key' });

const text = 'FastAPI is a modern, fast web framework for building APIs with Python 3.8+';

const result = await client.compress(text, 'auto');

console.log(`Original: ${result.originalTokens} tokens`);
console.log(`Compressed: ${result.compressedTokens} tokens`);
console.log(`Saved: ${result.tokensSaved} tokens (${((1-result.compressionRatio)*100).toFixed(1)}%)`);
console.log(`Time: ${result.compressionTimeMs.toFixed(0)}ms`);
console.log(`\nOriginal text:\n${result.originalText}`);
console.log(`\nCompressed text:\n${result.compressedText}`);

File: sdk/python-sdk/examples/basic_compression.py
"""
Basic compression example
"""

from concise import Concise

client = Concise(api_key="your-api-key")

text = "FastAPI is a modern, fast web framework for building APIs with Python 3.8+"

result = client.compress(text, level="auto")

print(f"Original: {result.original_tokens} tokens")
print(f"Compressed: {result.compressed_tokens} tokens")
print(f"Saved: {result.tokens_saved} tokens ({(1-result.compression_ratio)*100:.1f}%)")
print(f"Time: {result.compression_time_ms:.0f}ms")
print(f"\nOriginal text:\n{result.original_text}")
print(f"\nCompressed text:\n{result.compressed_text}")
